src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -1526,7 +1526,10 @@ private bool CompleteScriptBlockBody(Token lCurly, ref IScriptExtent bodyExtent,
 
                     UngetToken(rCurly);
                     endScriptBlock = bodyExtent ?? lCurly.Extent;
-                    ReportIncompleteInput(lCurly.Extent, rCurly.Extent, () => ParserStrings.MissingEndCurlyBrace);
+                    var errorExpr = calledFromNamedBlockRule
```

Comment:
Not needed in the PR, but I'd suggest removing my ugly hack of using LINQ expression trees and replace with `nameof(ParserStrings.MissingEndCurlyBrace)`.

When that code was written, `nameof` was not in C#, so the ugly and inefficient code was used because:
* It ensured no typos in the `FullyQualifiedErrorId`.
* It was easy to search for all references to an error message via the property.
* Performance was not deemed important because they were errors.

Performance might matter more than I originally assumed given editor scenarios - and given the `nameof` language feature, it's reasonable (but annoying and a little verbose) to change error reporting to look more like:

```C#
ReportIncompleteInput(lCurly.Extent, rCurly.Extent,
    ParserStrings.MissingEndCurlyBrace, nameof(ParserStrings.MissingEndCurlyBrace));
```

In other words, pass the error text **and** the error id.

Alternatively you could just pass the error id and use reflection of some sort to load the error string, but that would not be good for performance.

Reply:
Agreed. Open #6444 to track this work.